py-polars/polars/lazyframe/frame.py

```diff
@@ -1918,6 +1944,18 @@ def collect(
         if streaming:
             issue_unstable_warning("Streaming mode is considered unstable.")
 
+        is_gpu = (is_config_obj := isinstance(engine, GPUEngine)) or engine == "gpu"
```

Comment:
Can you validate this option so that only "gpu"/"cpu"/GPUEngine is allowed as input?